<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StageEditorWindow</class>
 <widget class="QMainWindow" name="StageEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>721</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SMB Level Workshop 2 Stage Editor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="WS2Editor::Widget::ViewportWidget" name="viewportWidget" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionImport"/>
    <addaction name="actionExport"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionDuplicate"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionSelectAll"/>
    <addaction name="actionFocusSelected"/>
    <addaction name="separator"/>
    <addaction name="actionNewNode"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuDebug">
     <property name="title">
      <string>Debug</string>
     </property>
     <addaction name="actionToggleViewResources"/>
    </widget>
    <addaction name="actionToggleViewOutliner"/>
    <addaction name="actionToggleViewProperties"/>
    <addaction name="actionToggleViewTimeline"/>
    <addaction name="menuDebug"/>
    <addaction name="separator"/>
    <addaction name="actionToggleViewToolbar"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionImport"/>
   <addaction name="actionExport"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionCut"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="separator"/>
   <addaction name="actionDelete"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="outlinerDockWidget">
   <property name="windowTitle">
    <string>Outliner</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="outlinerContentsWidget">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="WS2Editor::Widget::OutlinerWidget" name="outlinerTreeView">
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="propertiesDockWidget">
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="propertiesContentsWidget"/>
  </widget>
  <widget class="QDockWidget" name="timelineDockWidget">
   <property name="windowTitle">
    <string>Timeline</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="timelineContentsWidget"/>
  </widget>
  <widget class="QDockWidget" name="resourcesDockWidget">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Debug: Resources</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="resourcesContentsWidget">
    <layout class="QGridLayout" name="gridLayout_3">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QTableView" name="resourcesTableView">
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="textElideMode">
        <enum>Qt::ElideLeft</enum>
       </property>
       <property name="sortingEnabled">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>16</number>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/File.png</normaloff>:/Workshop2/Icons/File.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Folder.png</normaloff>:/Workshop2/Icons/Folder.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Close.png</normaloff>:/Workshop2/Icons/Close.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Undo.png</normaloff>:/Workshop2/Icons/Undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Redo.png</normaloff>:/Workshop2/Icons/Redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Cut.png</normaloff>:/Workshop2/Icons/Cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Copy.png</normaloff>:/Workshop2/Icons/Copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Paste.png</normaloff>:/Workshop2/Icons/Paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/SelectAll.png</normaloff>:/Workshop2/Icons/SelectAll.png</iconset>
   </property>
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionDuplicate">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Duplicate.png</normaloff>:/Workshop2/Icons/Duplicate.png</iconset>
   </property>
   <property name="text">
    <string>Duplicate</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Delete.png</normaloff>:/Workshop2/Icons/Delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionFocusSelected">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Focus.png</normaloff>:/Workshop2/Icons/Focus.png</iconset>
   </property>
   <property name="text">
    <string>Focus Selected</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Save.png</normaloff>:/Workshop2/Icons/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/SaveAs.png</normaloff>:/Workshop2/Icons/SaveAs.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionToggleViewToolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toolbar</string>
   </property>
  </action>
  <action name="actionToggleViewOutliner">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Outliner</string>
   </property>
  </action>
  <action name="actionToggleViewProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
  <action name="actionToggleViewTimeline">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Timeline</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Import.png</normaloff>:/Workshop2/Icons/Import.png</iconset>
   </property>
   <property name="text">
    <string>Import</string>
   </property>
   <property name="toolTip">
    <string>Import</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Export.png</normaloff>:/Workshop2/Icons/Export.png</iconset>
   </property>
   <property name="text">
    <string>Export</string>
   </property>
   <property name="toolTip">
    <string>Export</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionToggleViewResources">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Resources</string>
   </property>
  </action>
  <action name="actionNewNode">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Add.png</normaloff>:/Workshop2/Icons/Add.png</iconset>
   </property>
   <property name="text">
    <string>New Node</string>
   </property>
   <property name="toolTip">
    <string>Adds a new empty node to the scene</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../../resources/Workshop2/Workshop2Resources.qrc">
     <normaloff>:/Workshop2/Icons/Settings.png</normaloff>:/Workshop2/Icons/Settings.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>WS2Editor::Widget::OutlinerWidget</class>
   <extends>QTreeView</extends>
   <header>widget/OutlinerWidget.hpp</header>
  </customwidget>
  <customwidget>
   <class>WS2Editor::Widget::ViewportWidget</class>
   <extends>QWidget</extends>
   <header>widget/ViewportWidget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources/Workshop2/Workshop2Resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionToggleViewOutliner</sender>
   <signal>triggered(bool)</signal>
   <receiver>outlinerDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1151</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outlinerDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleViewOutliner</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1151</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>propertiesDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleViewProperties</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1151</x>
     <y>613</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleViewProperties</sender>
   <signal>triggered(bool)</signal>
   <receiver>propertiesDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1151</x>
     <y>613</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timelineDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleViewTimeline</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>678</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleViewTimeline</sender>
   <signal>triggered(bool)</signal>
   <receiver>timelineDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>678</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mainToolBar</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleViewToolbar</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>37</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleViewToolbar</sender>
   <signal>triggered(bool)</signal>
   <receiver>mainToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleViewResources</sender>
   <signal>triggered(bool)</signal>
   <receiver>resourcesDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resourcesDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleViewResources</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>354</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
